### PR TITLE
Add types for json-query

### PR DESCRIPTION
**Pull request by mtraynham**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not provide its own types, and you can not add them.
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by plantain-00 — APPROVED** [review]

**Comment by RyanCavanaugh** [comment]
Approved by third party. PR appears ready to merge pending review by a maintainer.

File: types/json-query/index.d.ts
Comment (DanielRosenwasser) [inline types/json-query/index.d.ts]: Maybe `[Selector] & { [x: number]: QueryParam }`

Comment (mtraynham) [reply types/json-query/index.d.ts]: I thought I tried that but came up empty handed.  Should that work?

**Pull request merged**
